migrations/versions/0398_add_failed_logins_table.py

```diff
@@ -0,0 +1,29 @@
+"""
+
+Revision ID: 0398_add_failed_logins_table
+Revises: 0397_drop_providers
+Create Date: 2024-06-19 16:31:49.657408
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects import postgresql
+
+revision = "0398_add_failed_logins_table"
+down_revision = "0397_drop_providers"
+
+
+def upgrade():
+    op.create_table(
+        "failed_logins",
+        sa.Column("id", postgresql.UUID(as_uuid=True), nullable=False),
+        sa.Column("ip", postgresql.INET(), nullable=False),
+        sa.Column("failed_login_count", sa.Integer(), nullable=False),
+        sa.Column("attempted_at", sa.DateTime(), nullable=False),
+        sa.PrimaryKeyConstraint("id"),
+    )
```

Comment:
Are we expecting to have multiple rows with the same IP address, if so how are we differentiating them?  You only seem to be querying by ip address, and this is also not an indexed column, which means that every query will perform a full table scan (this is very slow, you probably won't notice on a table which has very few rows, but once you get a thousand rows it will drastically slow down).

If we're not then why not lose the id column and have the ip address set as the primary key (this should/might also create an index for the column)?

If you're worried about the length of the column you could always make the column type a 4 byte integer and convert the IP into it's 4 byte representation.

At the very least you should create a unique constraint on the ip address if you don't want to have multiple rows with the same IP address.